docs/core/diagnostics/dotnet-trace.md

```diff
@@ -200,6 +203,22 @@ dotnet-trace collect [--buffersize <size>] [--clreventlevel <clreventlevel>] [--
 
   Shows the input and output streams of a launched child process in the current console.
 
+- **`--resume-runtime`**
+
+  Resume runtime once session has been initialized, defaults to true. Disable resume of runtime using --resume-runtime:false.
+
+- **`--stopping-event-provider-name`**
+
+  A string, parsed as-is, that will stop the trace upon hitting an event with the matching provider name. For a more specific stopping event, additionally provide `--stopping-event-event-name` and/or `--stopping-event-payload-filter`.
+
+- **`--stopping-event-event-name`**
+
+  A string, parsed as-is, that will stop the trace upon hitting an event with the matching event name. Requires `--stopping-event-provider-name` to be set. For a more specific stopping event, additionally provide `--stopping-event-payload-filter`.
+
+- **`--stopping-event-payload-filter`**
+
+  A string, parsed as [payload_field_name]:[payload_field_value] pairs separated by commas, that will stop the trace upon hitting an event with a matching payload. Requires `--stopping-event-provider-name` and `--stopping-event-event-name` to be set.
```

Comment:
If there are more than one name:value pairs, will the trace stop on the whichever one comes first?

Reply:
A `TraceEvent`'s payload typically will consist of multiple name/values.
For instance,
```
Hit StoppingEvent with ProviderName: `Microsoft-Windows-DotNETRuntime`  EventName: `Method/JittingStarted` at time: `10/24/2023 4:00:16 PM
Payload count: `8`
0 payloadName `MethodID` payloadValue `-5476376640355487248`
1 payloadName `ModuleID` payloadValue `-5476376641966429840`
2 payloadName `MethodToken` payloadValue `100664394`
3 payloadName `MethodILSize` payloadValue `68`
4 payloadName `MethodNamespace` payloadValue `System.AppContext`
5 payloadName `MethodName` payloadValue `Setup`
6 payloadName `MethodSignature` payloadValue `void(char**,char**,int)`
7 payloadName `ClrInstanceID` payloadValue `9`
```

[The filtering logic in `EventMonitor.cs`](https://github.com/dotnet/diagnostics/blob/bb25d231174007a23933dc1996bb752f8fbdabcd/src/Microsoft.Diagnostics.Monitoring.EventPipe/EventMonitor.cs#L99-L110) will only consider a match if all of the user provided pairs are a match.

If one were to pass `--stopping-event-payload-filter MethodNamespace:System.AppContext,MethodName:GetValue`, it wouldn't match the TraceEvent above.

Reply:
Would you please add in an example of using the new arguments? And have it clarify that all user provided pairs are a match?

Reply:
Done!

Reply:
Looks great to me!